File: Belajar Android/IntentActivity/app/src/main/java/com/umar/intentactivity/MainActivity.java
package com.umar.intentactivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private EditText etNamaBarang;

    private void load() {
        etNamaBarang = findViewById(R.id.etNamaBarang);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        load();
    }

    public void bukaBarangActivity(View view) {
        // Ambil teks dari EditText
        String namaBarang = etNamaBarang.getText().toString();

        // Buat Intent untuk berpindah ke BarangActivity
        Intent intent = new Intent(this, BarangActivity.class);

        // Kirim data menggunakan putExtra
        intent.putExtra("NAMA_BARANG_KEY", namaBarang);

        // Mulai Activity baru
        startActivity(intent);
    }
}

File: Projek Jetpack Compose/RealtimeWeather/app/src/main/java/com/umar/realtimeweather/MainActivity.kt
package com.umar.realtimeweather

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.ui.Modifier
import com.umar.realtimeweather.ui.WeatherPage
import com.umar.realtimeweather.ui.theme.RealtimeWeatherTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            RealtimeWeatherTheme {
                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
                    WeatherPage(modifier = Modifier.padding(innerPadding))
                }
            }
        }
    }
}
